src/common/TrackedOp.h

```diff
@@ -234,35 +234,26 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
 
   struct Event {
     utime_t stamp;
-    string str;
-    const char *cstr = nullptr;
+    std::string str;
```

Comment:
this forces us to create a `std::string` from `string_view`. IMHO, it practically defeats the purpose of using `string_view`. i'd use `std::variant<>` for holding the `string_view` or `std::string`.

Reply:
The `std::string_view` may not refer to static memory? In several places, an Event is created with `s` referring to a temporary string.

Reply:
i think we can differentiate that case at the caller site, can't we? and instead move the temporary string or copy it to the one in `Event`.

Reply:
I agree with you. See my comment here: https://github.com/ceph/ceph/pull/28849#discussion_r301836379